File: 036_Double-base_Palindromes/Solution.py
def convert_to_base(decimal_number, base):
    if decimal_number == 0:
        return 0

    digits = "0123456789"

    result = ""
    while decimal_number > 0:
        remainder = decimal_number % base
        result = digits[remainder] + result
        decimal_number //= base

    return result

def check_palindrome(num):
    return str(num) == str(num)[::-1]

n, k = map(int, input().strip().split())

_sum = 0

for num in range(1, n):
    base_repr = convert_to_base(num, k)
    if check_palindrome(num) and check_palindrome(base_repr):
        _sum += num
print(_sum)

File: 036_Double-base_Palindromes/Solution.go
package main

import (
	"fmt"
	"strconv"
)

func convertToBase(decimalNumber, base int) string {
	if decimalNumber == 0 {
		return "0"
	}

	digits := "0123456789"
	var result string
	for decimalNumber > 0 {
		remainder := decimalNumber % base
		result = string(digits[remainder]) + result
		decimalNumber /= base
	}
	return result
}

func checkPalindrome(num int) bool {
	numStr := strconv.Itoa(num)
	for i, j := 0, len(numStr)-1; i < j; i, j = i+1, j-1 {
		if numStr[i] != numStr[j] {
			return false
		}
	}
	return true
}

func main() {
	//Enter your code here. Read input from STDIN. Print output to STDOUT
	var n, k int
	fmt.Scan(&n, &k)

	var sum int
	for num := 1; num < n; num++ {
		baseRepr := convertToBase(num, k)
		baseReprInt, err := strconv.Atoi(baseRepr)
		if err != nil {
			fmt.Println("Error:", err)
			return
		}
		if checkPalindrome(num) && checkPalindrome(baseReprInt) {
			sum += num
		}
	}
	fmt.Println(sum)
}
